src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java

```diff
@@ -58,6 +61,200 @@ public static void exportAllHistoricalToFile(
+    }
+    if (includeTagGroupB) {
+      tagGroupFilterStr += "B";
+    }
+    if (includeTagGroupC) {
+      tagGroupFilterStr += "C";
+    }
+    if (includeTagGroupD) {
+      tagGroupFilterStr += "D";
+    }
+
+    // Get EBD data type
+    String ebdDataType;
+    if (stringHistorical) {
+      ebdDataType = "HS";
+    } else {
+      ebdDataType = "HL";
+    }
+
+    /*
+     * Build EBD string with parameters
+     * dt[ebdDataType]: data type, value specified in string ebdDataType
+     * ftT: file type, text
+     * startTime: start time for data
+     * endTime: end time for data
+     * flABCD: filter type, specified tag groups
+     */
+    return "$dt"
+        + ebdDataType
+        + "$ftT$st"
+        + startTime
+        + "$et"
+        + endTime
+        + "$fl"
+        + tagGroupFilterStr;
+  }
+
+  /**
+   * Parses Export Block Descriptor Historical Log response into dataPoints. Note: this function
+   * only handles Historical List responses
+   *
+   * @param exporter EBD Exporter
+   * @return dataPoints
+   * @throws IOException for parsing Exceptions
+   * @throws JSONException for JSON parsing Exceptions
+   * @throws TimeoutException timeout for reading event file
+   * @throws CircularizedFileException when event log indicates circularized event
+   */
+  private static ArrayList parseEBDHistoricalLogExportResponse(Exporter exporter)
+      throws IOException, JSONException, TimeoutException, CircularizedFileException {
+    final DataInputStream dataStream = new DataInputStream(exporter);
+    final BufferedReader reader = new BufferedReader(new InputStreamReader(dataStream));
+
+    // Create line counter and variable to store current line
+    int lineCount = 0;
+    String line = reader.readLine();
+    // Loop through lines in file until end and store data points
+    ArrayList dataPoints = new ArrayList();
+    while (line != null) {
+
+      // Only parse lines 1 and greater, skip header
+      if (lineCount > 0) {
+        // Parse line
+        DataPoint lineDataPoint = parseHistoricalFileLine(line);
+
+        if (lineDataPoint != null) {
+          dataPoints.add(lineDataPoint);
+        }
+      }
+
+      // Increment line count
+      lineCount++;
+      // Read next line before looping again
+      line = reader.readLine();
+    }
+    reader.close();
+    exporter.close();
+
+    // Check for Circularized Event
+    if (EventFile.DidFileCircularizedEventOccur()) {
+      throw new CircularizedFileException("A circularized event was found in the event logs.");
+    }
+    return dataPoints;
+  }
+
+  /**
+   * Executes EBD call, waits for data return.
+   *
+   * @param ebdStr EBD string
+   * @return EBD Exporter - caller must close exporter
+   * @throws TimeoutException when there is no response before timeout period
+   * @throws IOException when there is an Exporter Exception
+   */
+  public static Exporter ExecuteEBDCall(String ebdStr) throws IOException, TimeoutException {
```

Comment:
```suggestion
  public static Exporter ExecuteEbdCall(String ebdStr) throws IOException, TimeoutException {
```

Reply:
Shouldn't this be executeEbdCall() instead?

Reply:
changed